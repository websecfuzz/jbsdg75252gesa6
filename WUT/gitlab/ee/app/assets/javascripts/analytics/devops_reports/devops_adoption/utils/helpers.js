import { sprintf } from '~/locale';
import { joinPaths } from '~/lib/utils/url_utility';
import { GROUP_DEVOPS_PATH } from '../constants';

/**
 * A helper function which accepts the enabledNamespaces,
 *
 * @param {Object} params the enabledNamespaces data, and the check for open modals
 *
 * @return {Boolean} a boolean to determine if table data should be polled
 */
export const shouldPollTableData = ({ enabledNamespaces, openModal }) => {
  if (openModal) {
    return false;
  }
  if (!enabledNamespaces.length) {
    return true;
  }

  const anyPendingEnabledNamespaces = enabledNamespaces.some(
    (node) => node.latestSnapshot === null,
  );

  return anyPendingEnabledNamespaces;
};

/**
 * A helper function which extracts the total feature adoption count for a group
 * of snapshot data, filtered out by specific features / columns
 *
 * @param { Array } snapshots the snapshot data for a given group node
 * @param { Array } cols the columns which need to be used for the calculation
 *
 * @return { Array } an array containing the adopted counts for the given columns
 */
export const getAdoptedCountsByCols = (snapshots, cols) => {
  return snapshots.reduce((acc, snapshot) => {
    const adoptedCount = cols.reduce((adopted, col) => {
      return snapshot[col.key] ? adopted + 1 : adopted;
    }, 0);

    return [...acc, adoptedCount];
  }, []);
};

/**
 * A helper function which computes the DevOps Adoption feature path
 * given a specific group path
 *
 * @param { String } fullPath the full path for the group
 *
 * @return { String } the path for the group level DevOps Adoption feature
 */
export const getGroupAdoptionPath = (fullPath) => {
  const prefix = gon.relative_url_root || '';
  return fullPath ? joinPaths(prefix, sprintf(GROUP_DEVOPS_PATH, { fullPath })) : null;
};
