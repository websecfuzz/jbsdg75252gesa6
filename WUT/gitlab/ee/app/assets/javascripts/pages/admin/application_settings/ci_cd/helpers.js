/**
 * Filters [items] based on a given [searchTerm].
 * Catagories with no items after filtering are not included in the returned object.
 * @param {Object} allItems - { <categoryName>: [{ name, id }] }
 * @param {String} searchTerm
 * @returns {Object}
 */
export function filterItems(allItems, searchTerm) {
  return Object.entries(allItems)
    .map(([key, items]) => ({
      text: key,
      options: items
        .filter((item) => item.name.toLowerCase().includes(searchTerm))
        .map((item) => ({
          text: item.name,
          value: item.key,
        })),
    }))
    .filter((group) => group.options.length > 0);
}
