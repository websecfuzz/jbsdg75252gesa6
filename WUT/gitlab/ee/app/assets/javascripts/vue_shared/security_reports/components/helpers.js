import { __ } from '~/locale';
import { EMPTY_BODY_MESSAGE } from './constants';

/**
 * A helper function which validates the passed
 * in body string.
 *
 * It returns an empty string if the body has explicitly
 * been passed in as an empty string, a fallback
 * message if the body is null / undefined, else
 * it will return the original body string.
 *
 * @param {String} body the body message
 *
 * @return {String} the validated body message
 */
export const bodyWithFallBack = (body) => (body === '' ? '' : body || EMPTY_BODY_MESSAGE);

/**
 * Returns a string representation of the HTTP request
 *
 * @param {Object} httpData
 * @param {String} httpData.body
 * @param {String} httpData.method
 * @param {String} httpData.url
 * @param {{name: String, value: String}[]} httpData.headers
 *
 * @returns {String}
 */
export const getHttpString = (httpData) => {
  if (!httpData) {
    return '';
  }

  const { body, headers, method, url, statusCode, reasonPhrase } = httpData;
  const methodOrStatusCode = method || statusCode;
  const urlOrReasonPhrase = url || reasonPhrase;
  const headerString = headers.map(({ name, value }) => `${name}: ${value}`).join('\n');

  return `${methodOrStatusCode} ${urlOrReasonPhrase}\n${headerString}\n\n${bodyWithFallBack(body)}`;
};

export const getDismissalNoteEventText = ({ hasProject, hasPipeline, hasDismissalReason }) => {
  if (hasDismissalReason) {
    if (hasPipeline && hasProject) {
      return __(
        '%{statusStart}Dismissed%{statusEnd}: %{dismissalReason} on pipeline %{pipelineLink} at %{projectLink}',
      );
    }

    if (hasPipeline) {
      return __(
        '%{statusStart}Dismissed%{statusEnd}: %{dismissalReason} on pipeline %{pipelineLink}',
      );
    }

    if (hasProject) {
      return __('%{statusStart}Dismissed%{statusEnd}: %{dismissalReason} at %{projectLink}');
    }

    return __('%{statusStart}Dismissed%{statusEnd}: %{dismissalReason}');
  }

  if (hasPipeline && hasProject) {
    return __('%{statusStart}Dismissed%{statusEnd} on pipeline %{pipelineLink} at %{projectLink}');
  }

  if (hasPipeline) {
    return __('%{statusStart}Dismissed%{statusEnd} on pipeline %{pipelineLink}');
  }

  if (hasProject) {
    return __('%{statusStart}Dismissed%{statusEnd} at %{projectLink}');
  }

  return __('%{statusStart}Dismissed%{statusEnd}');
};
