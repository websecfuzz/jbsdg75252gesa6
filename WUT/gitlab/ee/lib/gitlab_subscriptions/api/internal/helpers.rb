# frozen_string_literal: true

module GitlabSubscriptions
  module API
    module Internal
      module Helpers
        def jwt_request?
          headers[Auth::INTERNAL_API_REQUEST_HEADER].present?
        end

        def authenticate_from_jwt!
          unauthorized! unless Auth.verify_api_request(headers)
        end
      end
    end
  end
end
