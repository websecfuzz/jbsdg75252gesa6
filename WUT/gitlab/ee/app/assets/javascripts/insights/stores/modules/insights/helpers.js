import { pairDataAndLabels, buildNullSeries } from 'ee/analytics/shared/utils';
import { BASE_NULL_SERIES_OPTIONS, BASE_SERIES_DATA_OPTIONS } from 'ee/analytics/shared/constants';
import {
  CHART_TYPES,
  INSIGHTS_DATA_SOURCE_DORA,
  INSIGHTS_NO_DATA_TOOLTIP,
} from 'ee/insights/constants';
import { __, s__ } from '~/locale';

const getGroupByValue = (queryParams) => {
  return queryParams.group_by || queryParams.params.group_by;
};

const getTypeValue = (queryParams) => {
  return queryParams.issuable_type || queryParams.params.issuable_type || queryParams.params.metric;
};

const getAxisTitle = (label) => {
  switch (label) {
    case 'day':
      return __('Days');
    case 'week':
      return __('Weeks');
    case 'month':
      return __('Months');
    case 'issue':
      return __('Issues');
    case 'merge_request':
      return __('Merge requests');
    case 'deployment_frequency':
      return s__('DORA4Metrics|Deployment frequency');
    case 'lead_time_for_changes':
      return s__('DORA4Metrics|Lead time for changes (median days)');
    case 'time_to_restore_service':
      return s__('DORA4Metrics|Time to restore service (median days)');
    case 'change_failure_rate':
      return s__('DORA4Metrics|Change failure rate (percentage)');
    default:
      return '';
  }
};

export const transformChartDataForGlCharts = (
  { type, query: queryParams },
  { labels, datasets },
) => {
  const yAxisTitle = getAxisTitle(getTypeValue(queryParams));
  const formattedData = {
    xAxisTitle: getAxisTitle(getGroupByValue(queryParams)),
    yAxisTitle,
    labels,
    datasets: [],
    seriesNames: [],
  };

  const { data_source: dataSource = '' } = queryParams;

  switch (type) {
    case CHART_TYPES.BAR:
      formattedData.datasets = [
        {
          name: 'all',
          data: labels.map((label, i) => [label, datasets[0].data[i]]),
        },
      ];
      break;
    case CHART_TYPES.STACKED_BAR:
      formattedData.datasets.push(
        ...datasets.map((dataset) => ({
          name: dataset.label,
          data: dataset.data,
        })),
      );
      break;
    case CHART_TYPES.LINE:
      if (dataSource === INSIGHTS_DATA_SOURCE_DORA) {
        const paired = pairDataAndLabels({
          datasetNames: [yAxisTitle],
          axisLabels: labels,
          datasets,
        });

        formattedData.datasets = buildNullSeries({
          seriesData: paired,
          seriesDataOptions: BASE_SERIES_DATA_OPTIONS,
          nullSeriesTitle: INSIGHTS_NO_DATA_TOOLTIP,
          nullSeriesOptions: BASE_NULL_SERIES_OPTIONS,
        });
      } else {
        formattedData.datasets.push(
          ...datasets.map((dataset) => ({
            name: dataset.label,
            data: labels.map((label, i) => [label, dataset.data[i]]),
          })),
        );
      }
      break;
    default:
      formattedData.datasets = { all: labels.map((label, i) => [label, datasets[0].data[i]]) };
  }
  return formattedData;
};

export default {
  transformChartDataForGlCharts,
};
