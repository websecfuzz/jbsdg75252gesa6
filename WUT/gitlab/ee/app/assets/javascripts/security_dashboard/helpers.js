import isPlainObject from 'lodash/isPlainObject';
import { REPORT_TYPES_ALL } from 'ee/security_dashboard/constants';
import { convertObjectPropsToSnakeCase } from '~/lib/utils/common_utils';
import { __ } from '~/locale';

/**
 * Provided a security reports summary from the GraphQL API, this returns an array of arrays
 * representing a properly formatted report ready to be displayed in the UI. Each sub-array consists
 * of the user-friend report's name, and the summary's payload. Note that summary entries are
 * considered empty and are filtered out of the return if the payload is `null` or don't include
 * a vulnerabilitiesCount property. Report types whose name can't be matched to a user-friendly
 * name are filtered out as well.
 *
 * Take the following summary for example:
 * {
 *   containerScanning: { vulnerabilitiesCount: 123 },
 *   invalidReportType: { vulnerabilitiesCount: 123 },
 *   dast: null,
 * }
 *
 * The formatted summary would look like this:
 * [
 *   ['containerScanning', { vulnerabilitiesCount: 123 }]
 * ]
 *
 * Note that `invalidReportType` was filtered out as it can't be matched with a user-friendly name,
 * and the DAST report was omitted because it's empty (`null`).
 *
 * @param {Object} rawSummary
 * @returns {Array}
 */
export const getFormattedSummary = (rawSummary = {}) => {
  if (!isPlainObject(rawSummary)) {
    return [];
  }
  // Convert keys to snake case so they can be matched against REPORT_TYPES keys for translation
  const snakeCasedSummary = convertObjectPropsToSnakeCase(rawSummary);
  // Convert object to an array of entries to make it easier to loop through
  const summaryEntries = Object.entries(snakeCasedSummary);
  // Filter out empty entries as we don't want to display those in the summary
  const withoutEmptyEntries = summaryEntries.filter(
    ([, scanSummary]) => scanSummary?.vulnerabilitiesCount !== undefined,
  );
  // Replace keys with translations found in REPORT_TYPES if available
  const formattedEntries = withoutEmptyEntries.map(([scanType, scanSummary]) => {
    const name = REPORT_TYPES_ALL[scanType];
    return name ? [name, scanSummary] : null;
  });
  // Filter out keys that could not be matched with any translation and are thus considered invalid
  return formattedEntries.filter((entry) => entry !== null);
};

/**
 * Limits the number of projects displayed per vulnerability grade.
 *
 * Takes an array of vulnerability grade data and ensures that each grade
 * shows at most the specified number of projects. Grades with fewer projects
 * than the limit are returned unchanged.
 *
 * @param {Array} vulnerabilityGradesQueryResults - Array of vulnerability grade objects from GraphQL query
 * @param {number} [maxProjects=5] - Maximum number of projects to show per grade
 * @returns {Array} Array of vulnerability grade objects with limited project nodes
 */
export const limitVulnerabilityGradeProjects = (
  vulnerabilityGradesQueryResults,
  maxProjects = 5,
) => {
  return vulnerabilityGradesQueryResults.map((gradeData) => {
    const nodes = gradeData.projects?.nodes;

    if (!nodes || nodes.length <= maxProjects) {
      return gradeData;
    }

    return {
      ...gradeData,
      projects: {
        ...gradeData.projects,
        nodes: nodes.slice(0, maxProjects),
      },
    };
  });
};

export const PROJECT_LOADING_ERROR_MESSAGE = __('An error occurred while retrieving projects.');

/**
 * Custom error class for PDF export operations that are not ready.
 *
 * @param {string} message - User-friendly error message describing why the export isn't ready
 */
export class PdfExportError extends Error {
  constructor(message) {
    super(message);
    this.name = 'PdfExportError';
  }
}

export default () => ({});
