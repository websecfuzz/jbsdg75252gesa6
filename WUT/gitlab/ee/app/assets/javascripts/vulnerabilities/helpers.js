import { getIdFromGraphQLId } from '~/graphql_shared/utils';
import { isAbsolute, isValidURL } from '~/lib/utils/url_utility';
import {
  REGEXES,
  SUPPORTED_IDENTIFIER_TYPE_CWE,
  SUPPORTED_IDENTIFIER_TYPE_OWASP,
  VULNERABILITY_TAB_INDEX_TO_NAME,
} from './constants';

// Get the issue in the format expected by the descendant components of related_issues_block.vue.
export const getFormattedIssue = (issue) => ({
  ...issue,
  reference: `#${issue.iid}`,
  path: issue.web_url,
});

export const getAddRelatedIssueRequestParams = (reference, defaultProjectId) => {
  let issueId = reference;
  let projectId = defaultProjectId;

  // If the reference is an issue number, parse out just the issue number.
  if (REGEXES.ISSUE_FORMAT.test(reference)) {
    [, issueId] = REGEXES.ISSUE_FORMAT.exec(reference);
  }
  // If the reference is an absolute URL and matches the issues URL format, parse out the project and issue.
  else if (isValidURL(reference) && isAbsolute(reference)) {
    const { pathname } = new URL(reference);

    if (REGEXES.LINK_FORMAT.test(pathname)) {
      [, projectId, issueId] = REGEXES.LINK_FORMAT.exec(pathname);
    }
  }

  return { target_issue_iid: issueId, target_project_id: projectId };
};

export const normalizeGraphQLNote = (note) => {
  if (!note) {
    return null;
  }

  return {
    ...note,
    id: getIdFromGraphQLId(note.id),
    author: {
      ...note.author,
      id: getIdFromGraphQLId(note.author.id),
      path: note.author.webPath,
    },
  };
};

export const normalizeGraphQLVulnerability = (vulnerability) => {
  if (!vulnerability) {
    return null;
  }

  const newVulnerability = { ...vulnerability };

  if (vulnerability.id) {
    newVulnerability.id = getIdFromGraphQLId(vulnerability.id);
  }

  if (vulnerability.state) {
    newVulnerability.state = vulnerability.state.toLowerCase();
  }

  ['confirmed', 'resolved', 'dismissed'].forEach((state) => {
    if (vulnerability[`${state}By`]?.id) {
      newVulnerability[`${state}ById`] = getIdFromGraphQLId(vulnerability[`${state}By`].id);
      delete newVulnerability[`${state}By`];
    }
  });

  return newVulnerability;
};

export const normalizeGraphQLLastStateTransition = (graphQLVulnerability, vulnerability) => {
  const stateTransitions = [...vulnerability.stateTransitions];

  // The vulnerability status mutation only returns 1 stateTransition
  const [graphQLLastStateTransitions] = graphQLVulnerability.stateTransitions.nodes;
  stateTransitions.push({
    ...graphQLLastStateTransitions,
    dismissalReason: graphQLLastStateTransitions.dismissalReason?.toLowerCase(),
  });

  return { stateTransitions };
};

export const formatIdentifierExternalIds = ({ externalType, externalId, name }) => {
  return `[${externalType}]-[${externalId}]-[${name}]`;
};

export const isSupportedIdentifier = (externalType) => {
  return (
    externalType?.toLowerCase() === SUPPORTED_IDENTIFIER_TYPE_CWE ||
    // Case matters here. owasp and OWASP require different configuration
    // Currently, our API only support lowercase owasp
    // Uppercase OWASP will be supported in a follow up issue:
    // https://gitlab.com/gitlab-org/gitlab/-/issues/366556
    externalType === SUPPORTED_IDENTIFIER_TYPE_OWASP
  );
};

export const getRefFromBlobPath = (path) => {
  // Matches the 40-character hex string of the Git SHA in the blob path, for example:
  // "/group/project/-/blob/cdeda7ae724a332e008d17245209d5edd9ba6499/src/file.js"
  // will match "cdeda7ae724a332e008d17245209d5edd9ba6499".
  if (typeof path === 'string') {
    const match = path.match(/blob\/([a-f0-9]{40})/);
    return match ? match[1] : '';
  }
  return '';
};

/**
 *  Determines the tab index for code flow based on the route query parameters.
 */
export const getTabIndexForCodeFlowPage = (route) => {
  if (route?.query?.tab) {
    return 1;
  }
  return 0;
};

/**
 * @typedef {Object} TabNavigationParams
 * @property {string} path - The current route path.
 * @property {number} index - The index of the tab to be set.
 * @property {number} [selectedIndex] - The currently selected tab index. This is optional as it might be undefined.
 */

/**
 * Updates the route to reflect the selected tab index.
 * This function is used to update the URL when switching between different views
 * of the vulnerability, between 'details' and 'code flow'
 * @param {Object} router - The router object used for navigation.
 * @param {TabNavigationParams} routeProps - The route properties including path and tab indices.
 */
export const setTabIndexForCodeFlowPage = (router, routeProps) => {
  const { path, index, selectedIndex } = routeProps;

  // Don't set the tab index if it's the same. This prevents some querystring bugs.
  if (index === selectedIndex) {
    return;
  }
  router.push({
    path,
    query: { tab: VULNERABILITY_TAB_INDEX_TO_NAME[index] },
  });
};
