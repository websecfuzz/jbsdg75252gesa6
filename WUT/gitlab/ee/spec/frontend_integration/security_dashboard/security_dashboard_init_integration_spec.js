import initSecurityDashboard from 'ee/security_dashboard/security_dashboard_init';
import {
  DASHBOARD_TYPE_GROUP,
  DASHBOARD_TYPE_INSTANCE,
  DASHBOARD_TYPE_PROJECT,
} from 'ee/security_dashboard/constants';
import setWindowLocation from 'helpers/set_window_location_helper';
import { TEST_HOST } from 'helpers/test_constants';

const EMPTY_DIV = document.createElement('div');

const TEST_DATASET = {
  link: '/test/link',
  svgPath: '/test/no_changes_state.svg',
  emptyStateSvgPath: '/test/empty_state.svg',
  hasProjects: 'true',
};

describe('Security Dashboard', () => {
  let vm;
  let root;

  beforeEach(() => {
    root = document.createElement('div');
    document.body.appendChild(root);

    setWindowLocation(`${TEST_HOST}/-/security/dashboard`);

    // We currently have feature flag logic that needs gon.features to be set
    // It is set to false by default, so the original test's (where there was no feature flag) snapshot is preserved.
    window.gon.features = {
      groupSecurityDashboardNew: false,
    };
  });

  afterEach(() => {
    if (vm) {
      vm.$destroy();
    }
    root.remove();
  });

  const createComponent = ({ data, type }) => {
    const el = document.createElement('div');
    Object.assign(el.dataset, { ...TEST_DATASET, ...data });
    root.appendChild(el);
    vm = initSecurityDashboard(el, type);
  };

  const createEmptyComponent = () => {
    vm = initSecurityDashboard(null, null);
  };

  describe('default states', () => {
    it('sets up group-level', () => {
      createComponent({ data: { groupFullPath: '/test/' }, type: DASHBOARD_TYPE_GROUP });

      expect(root).toMatchSnapshot();
    });

    it('sets up group-level with `groupSecurityDashboardNew` feature flag set to `true`', () => {
      window.gon.features.groupSecurityDashboardNew = true;

      createComponent({ data: { groupFullPath: '/test/' }, type: DASHBOARD_TYPE_GROUP });

      expect(root).toMatchSnapshot();
    });

    it('sets up instance-level', () => {
      createComponent({
        data: { instanceDashboardSettingsPath: '/instance/settings_page' },
        type: DASHBOARD_TYPE_INSTANCE,
      });

      expect(root).toMatchSnapshot();
    });

    it('sets up project-level', () => {
      createComponent({
        data: {
          projectFullPath: '/test/project',
          hasVulnerabilities: 'true',
          securityConfigurationPath: '/test/configuration',
        },
        type: DASHBOARD_TYPE_PROJECT,
      });

      expect(root).toMatchSnapshot();
    });
  });

  describe('error states', () => {
    it('does not have an element', () => {
      createEmptyComponent();

      expect(root).toStrictEqual(EMPTY_DIV);
    });

    it('has unavailable pages', () => {
      createComponent({ data: { isUnavailable: true } });

      expect(root).toMatchSnapshot();
    });
  });
});
