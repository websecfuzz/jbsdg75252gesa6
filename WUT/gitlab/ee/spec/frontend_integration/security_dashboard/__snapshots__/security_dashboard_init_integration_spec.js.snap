// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Security Dashboard default states sets up group-level 1`] = `
<div>
  <div>
    <header
      class="gl-flex gl-flex-wrap gl-gap-y-3 gl-items-center gl-justify-between gl-my-5"
    >
      <div
        class="gl-flex gl-flex-wrap gl-gap-x-5 gl-gap-y-3 gl-justify-between gl-w-full md:gl-flex-nowrap"
      >
        <h1
          class="!gl-m-0 gl-heading-1"
          data-testid="page-heading"
        >
          Security dashboard
        </h1>
        <div
          class="gl-flex gl-flex-wrap gl-gap-3 gl-items-center gl-shrink-0 gl-w-full lg:gl-mt-2 md:gl-mt-1 page-heading-actions sm:gl-w-auto"
          data-testid="page-heading-actions"
        />
      </div>
    </header>
    <div
      class="gl-grid security-charts"
    >
      <section
        class="gl-border-1 gl-border-default gl-border-solid gl-flex gl-flex-col gl-rounded-base"
      >
        <header
          class="gl-p-5"
        >
          <h4
            class="gl-mb-3 gl-mt-0"
            data-testid="title"
          >
            Vulnerabilities over time
          </h4>
          <p
            class="gl-m-0 gl-text-subtle"
            data-testid="help-text"
          >
            June 6th to today
          </p>
          <div
            class="gl-mt-5"
            data-testid="controls"
          >
            <div
              class="btn-group gl-button-group gl-flex"
              role="group"
            >
              <button
                class="btn btn-default btn-md gl-button selected"
                type="button"
              >
                <span
                  class="gl-button-text"
                >
                  30 Days
                </span>
              </button>
              <button
                class="btn btn-default btn-md gl-button"
                type="button"
              >
                <span
                  class="gl-button-text"
                >
                  60 Days
                </span>
              </button>
              <button
                class="btn btn-default btn-md gl-button"
                type="button"
              >
                <span
                  class="gl-button-text"
                >
                  90 Days
                </span>
              </button>
            </div>
          </div>
        </header>
        <div
          aria-label="Loading"
          class="gl-my-12 gl-spinner-container"
          role="status"
        >
          <span
            class="!gl-align-text-bottom gl-spinner gl-spinner-dark gl-spinner-lg"
          />
        </div>
      </section>
      <section
        class="gl-border-1 gl-border-default gl-border-solid gl-flex gl-flex-col gl-rounded-base"
      >
        <header
          class="gl-p-5"
        >
          <h4
            class="gl-mb-3 gl-mt-0"
            data-testid="title"
          >
            Project security status
          </h4>
          <p
            class="gl-m-0 gl-text-subtle"
            data-testid="help-text"
          />
        </header>
        <div
          aria-label="Loading"
          class="gl-my-12 gl-spinner-container"
          role="status"
        >
          <span
            class="!gl-align-text-bottom gl-spinner gl-spinner-dark gl-spinner-lg"
          />
        </div>
      </section>
    </div>
  </div>
</div>
`;

exports[`Security Dashboard default states sets up group-level with \`groupSecurityDashboardNew\` feature flag set to \`true\` 1`] = `
<div>
  <div
    data-testid="security-dashboard-new"
  >
    <section
      class="gl-flex gl-items-center gl-my-4"
    >
      <div
        class="gl-flex gl-flex-col gl-w-full"
      >
        <div
          class="gl-flex gl-items-center"
        >
          <h2
            class="gl-my-0"
            data-testid="title"
          >
            Security dashboard
          </h2>
        </div>
        <div
          class="gl-flex gl-mt-3"
        >
          <p
            class="gl-mb-0"
            data-testid="description"
          >
            This dashboard provides an overview of your security vulnerabilities.
          </p>
        </div>
      </div>
    </section>
    <div
      class="gl-flex"
    >
      <div
        class="gl-flex gl-flex-col gl-grow"
      >
        <section
          class="gl-flex gl-flex-row gl-flex-wrap gl-gap-5 gl-pb-3 gl-pt-4"
          data-testid="filters-container"
        >
          <div
            class="gl-search-box-by-click gl-search-box-by-click-with-search-button input-group"
            data-testid="filtered-search-input"
            role="group"
          >
            <div
              class="gl-filtered-search-scrollable-container gl-filtered-search-scrollable-container-with-search-button"
            >
              <div
                class="gl-filtered-search-scrollable"
              >
                <div
                  class="gl-filtered-search-item gl-filtered-search-last-item gl-filtered-search-term gl-h-auto"
                  data-testid="filtered-search-term"
                >
                  <div
                    class="gl-filtered-search-term-token gl-filtered-search-token-segment"
                    data-testid="filtered-search-token-segment"
                  >
                    <input
                      aria-label="Search"
                      class="gl-filtered-search-term-input"
                      data-testid="filtered-search-term-input"
                      placeholder="Filter results..."
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="input-group-append"
            >
              <button
                aria-label="Search"
                class="btn btn-default btn-default-tertiary btn-icon btn-md gl-button gl-search-box-by-click-search-button"
                data-testid="search-button"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="gl-button-icon gl-fill-current gl-icon s16"
                  data-testid="search-icon"
                  role="img"
                >
                  <use
                    href="file-mock#search"
                  />
                </svg>
              </button>
            </div>
          </div>
        </section>
        <div
          class="-gl-mx-3 grid-stack"
          data-testid="gridstack-grid"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Security Dashboard default states sets up instance-level 1`] = `
<div>
  <div>
    <header
      class="gl-flex gl-flex-wrap gl-gap-y-3 gl-items-center gl-justify-between gl-my-5"
    >
      <div
        class="gl-flex gl-flex-wrap gl-gap-x-5 gl-gap-y-3 gl-justify-between gl-w-full md:gl-flex-nowrap"
      >
        <h1
          class="!gl-m-0 gl-heading-1"
          data-testid="page-heading"
        >
          Security dashboard
        </h1>
        <div
          class="gl-flex gl-flex-wrap gl-gap-3 gl-items-center gl-shrink-0 gl-w-full lg:gl-mt-2 md:gl-mt-1 page-heading-actions sm:gl-w-auto"
          data-testid="page-heading-actions"
        />
      </div>
    </header>
    <div
      class="gl-grid security-charts"
    >
      <section
        class="gl-border-1 gl-border-default gl-border-solid gl-flex gl-flex-col gl-rounded-base"
      >
        <header
          class="gl-p-5"
        >
          <h4
            class="gl-mb-3 gl-mt-0"
            data-testid="title"
          >
            Vulnerabilities over time
          </h4>
          <p
            class="gl-m-0 gl-text-subtle"
            data-testid="help-text"
          >
            June 6th to today
          </p>
          <div
            class="gl-mt-5"
            data-testid="controls"
          >
            <div
              class="btn-group gl-button-group gl-flex"
              role="group"
            >
              <button
                class="btn btn-default btn-md gl-button selected"
                type="button"
              >
                <span
                  class="gl-button-text"
                >
                  30 Days
                </span>
              </button>
              <button
                class="btn btn-default btn-md gl-button"
                type="button"
              >
                <span
                  class="gl-button-text"
                >
                  60 Days
                </span>
              </button>
              <button
                class="btn btn-default btn-md gl-button"
                type="button"
              >
                <span
                  class="gl-button-text"
                >
                  90 Days
                </span>
              </button>
            </div>
          </div>
        </header>
        <div
          aria-label="Loading"
          class="gl-my-12 gl-spinner-container"
          role="status"
        >
          <span
            class="!gl-align-text-bottom gl-spinner gl-spinner-dark gl-spinner-lg"
          />
        </div>
      </section>
      <section
        class="gl-border-1 gl-border-default gl-border-solid gl-flex gl-flex-col gl-rounded-base"
      >
        <header
          class="gl-p-5"
        >
          <h4
            class="gl-mb-3 gl-mt-0"
            data-testid="title"
          >
            Project security status
          </h4>
          <p
            class="gl-m-0 gl-text-subtle"
            data-testid="help-text"
          />
        </header>
        <div
          aria-label="Loading"
          class="gl-my-12 gl-spinner-container"
          role="status"
        >
          <span
            class="!gl-align-text-bottom gl-spinner gl-spinner-dark gl-spinner-lg"
          />
        </div>
      </section>
    </div>
  </div>
</div>
`;

exports[`Security Dashboard default states sets up project-level 1`] = `
<div>
  <div>
    <header
      class="gl-flex gl-flex-wrap gl-gap-y-3 gl-items-center gl-justify-between gl-my-5"
    >
      <div
        class="gl-flex gl-flex-wrap gl-gap-x-5 gl-gap-y-3 gl-justify-between gl-w-full md:gl-flex-nowrap"
      >
        <h1
          class="!gl-m-0 gl-heading-1"
          data-testid="page-heading"
        >
          Security dashboard
        </h1>
        <div
          class="gl-flex gl-flex-wrap gl-gap-3 gl-items-center gl-shrink-0 gl-w-full lg:gl-mt-2 md:gl-mt-1 page-heading-actions sm:gl-w-auto"
          data-testid="page-heading-actions"
        />
      </div>
      <div
        class="gl-text-subtle gl-w-full"
        data-testid="page-heading-description"
      >
        Historical view of open vulnerabilities in the default branch. Excludes vulnerabilities that were resolved or dismissed.
        <a
          class="gl-link"
          href="/help/user/application_security/security_dashboard/_index#project-security-dashboard"
          rel="noopener noreferrer"
          target="_blank"
        >
          Learn more.
        </a>
      </div>
    </header>
    <div>
      <div
        aria-label="Loading"
        class="gl-mt-6 gl-spinner-container"
        role="status"
      >
        <span
          class="!gl-align-text-bottom gl-spinner gl-spinner-dark gl-spinner-lg"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Security Dashboard error states has unavailable pages 1`] = `
<div>
  <section
    class="gl-empty-state gl-flex gl-flex-col gl-text-center"
  >
    <div
      class="gl-max-w-full"
    >
      <img
        alt=""
        class="gl-dark-invert-keep-hue gl-max-w-full"
        src="/test/empty_state.svg"
      />
    </div>
    <div
      class="gl-empty-state-content gl-m-auto gl-mx-auto gl-my-0 gl-p-5"
      data-testid="gl-empty-state-content"
    >
      <h1
        class="gl-leading-36 gl-mb-0 gl-mt-0 gl-text-size-h-display h4"
      >
        Oops, something doesn't seem right.
      </h1>
      <p
        class="gl-mb-0 gl-mt-4 gl-text-subtle"
      >
        Either you don't have permission to view this dashboard or the dashboard has not been setup. Please check your permission settings with your administrator or check your dashboard configurations to proceed.
      </p>
      <div
        class="gl-flex gl-flex-wrap gl-justify-center gl-mt-5"
      >
        <a
          class="btn btn-confirm btn-md gl-button gl-mb-3 gl-mx-2"
          href="/help/user/application_security/detect/security_configuration#with-the-ui"
        >
          <span
            class="gl-button-text"
          >
            Learn more about setting up your dashboard
          </span>
        </a>
      </div>
    </div>
  </section>
</div>
`;
